tests/test_pause_overlay.py

```
@@ -0,0 +1,443 @@
+    """Test suite for the PauseOverlay class."""
+
+    def test_init_creates_invisible_overlay(self, mock_pygame, mock_screen):
+        """PauseOverlay should initialize as invisible."""
+        overlay = PauseOverlay(mock_screen)
+
+        assert overlay.screen == mock_screen
+        assert overlay.visible is False
+
+    def test_init_creates_fonts(self, mock_pygame, mock_screen):
+        """PauseOverlay should create large and small fonts."""
+        overlay = PauseOverlay(mock_screen)
```

Ruff (0.15.2)

[error] Local variable `overlay` is assigned to but never used

Remove assignment to unused variable `overlay`

(F841)